src/inspector_socket.cc

```diff
@@ -48,16 +49,17 @@ static void dump_hex(const char* buf, size_t len) {
 }
 #endif
 
+static void remove_from_beginning(std::vector<char>* buffer, size_t count) {
```

Comment:
nit: I’m fine with having this the way it is, but to me using pointers to STL containers always seems a bit unidiomatic – this seems like a very okay place to use references?